### PR TITLE
Add credential checking logic when auth params are provided in wrangler config

### LINKED ISSUE #439: wrangler config should test for valid credentials
#### Environment

* operating system:
* output of `rustc -V`: `rustc 1.35.0 (3c235d560 2019-05-20)`
* output of `node -v`: `v12.4.0`
* output of `wrangler -V`: `👷 ✨   wrangler 1.1.1`

#### Steps to reproduce

Leave email + api key fields blank when running `wrangler config`

```bash
❯ wrangler whoami
💁‍  👋  You are logged with the email 'me@cloudflare.com'.
❯ wrangler config
Enter email: 

Enter api key: 

✨   Successfully configured. You can find your configuration file at: /Users/me/.wrangler/config/default.toml
❯ wrangler whoami
💁‍  👋  You are logged with the email ''.
❯ rm /Users/me/.wrangler/config/default.toml 
❯ wrangler whoami
Error: ⚠️  Your global config has an error, run `wrangler config`: missing field `email`
```

#### What did you expect to see?

I either expect to see `wrangler config` return an error when I pass it invalid credentials, and for `wrangler whoami` to error when there are invalid credentials in their `.wrangler/config/default.toml`

#### What did you see instead?

No errors and `wrangler whoami` saying that I am successfully logged in as `email ''`.

#### Other cases

It's not just empty credentials that can be successfully configured, completely fake ones work just fine as well.

```bash
❯ wrangler config
Enter email: 
a@cloudflare.com
Enter api key: 
1234
✨   Successfully configured. You can find your configuration file at: /Users/me/.wrangler/config/default.toml
❯ wrangler whoami
💁‍  👋  You are logged with the email 'a@cloudflare.com'.
```

### PR DESCRIPTION
**Pull request by gabbifish**
Closes #439 

Note: This PR will fail to build until https://github.com/cloudflare/cloudflare-rs/pull/58 and https://github.com/cloudflare/cloudflare-rs/pull/59 are merged into cloudflare-rs and a cloudflare-rs release is cut. This should also only be merged after #841 is merged.

This PR validates credentials given in `wrangler config` by testing them against the Cloudflare API. If credentials are invalid, users are asked to check their token or email/api key pair and make sure they input what they expected.

Example output:
```console
$ target/debug/wrangler config 
Enter API token: 
<fake-token>
 Verifying that provided credentials are valid...
Error: Credential validity check failed. Please check your API token. 
 Code 1000: Invalid API Token
```
^ no emoji because linux :(

### CODE REVIEW

File: src/commands/config/mod.rs [hunk 1]
```diff
@@ -37,3 +46,30 @@ pub fn global_config(user: &GlobalUser) -> Result<(), failure::Error> {
...
+                        failure::bail!("Auth check failed. Your token has status \"{}\", not \"active\".")
+                    }
+                },
+                Err(e) => failure::bail!("Auth check failed. Please make sure your API token is correct. \n{}", http::format_error(e, ErrorCodeDetail::None))
+            }
+        }
+        GlobalUser::GlobalKeyAuth { .. } => {
+            match client.request(&GetUserDetails {}) {
+                Ok(_) => Ok(()),
+                Err(e) => failure::bail!("Auth check failed. Please make sure your email and global API key pair are correct. \n{}", http::format_error(e, ErrorCodeDetail::None)),
```
[the remark this answers is not in the document]
Comment (ashleymichal) [reply src/commands/config/mod.rs]: we should have something in the workers wrangler quick start. if it doesn't already exist, we should update it to include api token instructions as part of this project

[the remark this answers is not in the document]
Comment (ashleymichal) [reply src/commands/config/mod.rs]: https://developers.cloudflare.com/workers/quickstart/#global-api-key

[the remark this answers is not in the document]
Comment (ashleymichal) [reply src/commands/config/mod.rs]: i think the above link is the most stable atm

[the remark this answers is not in the document]
Comment (gabbifish) [reply src/commands/config/mod.rs]: The link above seems to have outdated documentation, I'm going to open up a workers docs ticket to fix that (as well as add a section for api tokens). I'll omit the link from Wrangler for now, but will add them when the docs are ready!

**Comment by gabbifish** [comment]
An update to this PR: I also add support for a `--no-verify` flag that can be passed into `wrangler config` so that users can skip the config step if they are in place without internet (unlikely). The real reason for adding this flag was to support full invocation testing in tests/config.rs; calling the external validation endpoints would not work with the dummy credentials provided, so the test would always fail. 

Since this test calls `wrangler config` directly, we can't mock the validation endpoint. This left me to think providing the new flag above was the best option we have.

There is a world, however, where we could make a testing account for wrangler and provide its credentials... but we'd be storing those as plaintext in our tests, which is not good from a security practice.

File: src/commands/config/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ashleymichal) [reply src/commands/config/mod.rs]: i think it is fine to include the link, given the location will not change, only the content, and the content must be updated before we release.

**Comment by EverlastingBugstopper** [comment]
Flag seems great! FWIW I think we would want to make mocks for our API calls before we would make a testing account.

**Comment by ashleymichal** [comment]
yeah i'd be more interested in writing unit tests with mocked api clients; integration tests are important, but i always question including api requests in fully-automated CI test suites. this is an ongoing struggle for me.

File: src/commands/config/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (gabbifish) [reply src/commands/config/mod.rs]: I'm game to change the link to be a general https://developers.cloudflare.com/workers/quickstart/#authentication link (note: this endpoint currently doesn't exist). This link would contain information about both API tokens and API keys... what do you think? In the meantime I'll add the existing link.

[the remark this answers is not in the document]
Comment (ashleymichal) [reply src/commands/config/mod.rs]: ah that seems like the way to go. yeah add the existing link, write an issue for us here, and an accompanying issue for docs (cc @victoriabernard92 ) and this is good to merge.

**Pull request merged**